pkg/volume/csi/csi_plugin.go

```diff
@@ -214,7 +215,24 @@ func (p *csiPlugin) Init(host volume.VolumeHost) error {
 			}
 		}
 	}
-
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) {
+		csiClient := host.GetKubeClient()
+		if csiClient == nil {
+			klog.Warning(log("kubeclient not set, assuming standalone kubelet"))
+		} else {
+			adcHost, ok := host.(volume.AttachDetachVolumeHost)
```

Comment:
we should not need CSINode in AttachDetach controller. The only valid place where it should be initialized is KubeletVolumeHost. In fact, we should only call `NewNodeInfoManager` if KubeletVolumeHost is valid.